jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -1074,6 +1074,27 @@ def test_errors(self):
         "Shapes must be 1D sequences of concrete values of integer type"):
       core.dimension_as_value(np.float32(1))
 
+  def test_error_jax_value(self):
+
+    x = np.ones([3, 5], dtype=np.float32)
+
+    with self.assertRaisesRegex(
+        core.InconclusiveDimensionOperation,
+        "Dimension polynomial 'b' used in a context that requires a constant"):
+      self.CheckShapePolymorphism(
+          # Cannot use dimension variable as a JAX value
+          lambda x: jnp.array(x.shape[0]),
```

Comment:
How about `jnp.shape(x)[0]`? Is it the same behavior as `x.shape[0]` in JAX, or different?

Reply:
`jnp.shape` is synonim for np.shape : https://github.com/google/jax/blob/3a949acccb47f6b6d22c71896569d047c0590a88/jax/_src/numpy/lax_numpy.py#L115

so `x.shape` and `jnp.shape` would produce the same result. In general, for computations that should happen at tracing time we use the regular `np`.